components/Review/ReviewModal.tsx

```diff
@@ -0,0 +1,279 @@
+  return (
+    <Pressable
+      style={styles.navoptionpressable}
+      onPress={() => navaction(index)}
+    >
+      <Text style={styles.navoptiontext}>{title}</Text>
+      <Entypo name="chevron-right" size={25} />
+    </Pressable>
+  );
+}
+
+interface ModalProps {
+  modalVisibilitySetter: Dispatch<boolean>;
+  nextScreenIncr: Dispatch<number>;
+}
+
+function OptionsView({ modalVisibilitySetter, nextScreenIncr }: ModalProps) {
+  return (
+    <View style={styles.modalcontainer}>
+      <View style={styles.modalView}>
+        <ModalNavOption
+          index={1}
+          title="Edit Review"
+          navaction={nextScreenIncr}
+        />
+        <ModalNavOption
+          index={2}
+          title="Delete Review"
+          navaction={nextScreenIncr}
+        />
+
+        <Pressable
+          style={{ alignItems: "center", padding: 5 }}
+          onPress={() => modalVisibilitySetter(false)}
+        >
+          <Text style={styles.modalcanceltext}>Cancel</Text>
+        </Pressable>
+      </View>
+    </View>
+  );
+}
+
+interface DeleteReviewProps {
+  modalVisibilitySetter: Dispatch<boolean>;
+  nextScreenIncr: Dispatch<number>;
+}
+
+function DeleteView({
+  modalVisibilitySetter,
+  nextScreenIncr,
+}: DeleteReviewProps) {
+  return (
+    <View style={styles.modalcontainer}>
+      <View style={styles.modalView}>
+        <Text
+          style={{ fontFamily: "Poppins-Regular", fontSize: 14, opacity: 0.5 }}
+        >
+          Confirm Deletion of this Review
+        </Text>
+        <View
+          style={{
+            flexDirection: "row",
+            justifyContent: "center",
+            alignItems: "center",
+          }}
+        >
+          <Pressable
+            style={{ marginHorizontal: 10 }}
+            onPress={() => {
+              nextScreenIncr(0);
+              modalVisibilitySetter(false);
+            }}
+          >
+            <Text style={styles.modalcanceltext}>Cancel</Text>
+          </Pressable>
+          <Pressable
+            style={{
+              marginHorizontal: 10,
+              paddingHorizontal: 20,
+              paddingVertical: 10,
+              borderRadius: 25,
+              backgroundColor: "#FA4A0C",
+            }}
+            onPress={() => {
+              // 1. get all review data for user with: useAppSelector(selectAllReviews);
+              // 2. pass this data into flatlist
+              // 3. for each item in the flatlist, pass the review data
+              // 4. pass ID for current review along to "delete review"
+              // 5. on delete, call reducer to filter out current review ID
+            }}
+          >
+            <Text style={styles.sendreqbutton}>Delete</Text>
+          </Pressable>
+        </View>
+      </View>
+    </View>
+  );
+}
+
+interface EditReviewProps {
+  modalVisibilitySetter: Dispatch<boolean>;
+  nextScreenIncr: Dispatch<number>;
+}
+
+function EditView({ modalVisibilitySetter, nextScreenIncr }: EditReviewProps) {
```

Comment:
Eventually I think the Edit Review button will go to [this page](https://www.figma.com/file/D841mXwjr6o0VvnA2MSEq8/Food-delivery-app-Ui-kit-(Community)?node-id=1561%3A26407). What you made now works great for this quarter though!